core/domain/exp_domain.py

```diff
@@ -1298,6 +1316,26 @@ def is_demo(self):
         """
         return self.is_demo_exploration_id(self.id)
 
+    @property
+    def proto_size_in_bytes(self):
+        """Whether the exploration proto size is up to date.
+
+        Returns:
+            int. Updated exploration's proto size.
+        """
+        if self.android_proto_size_is_stale:
+            self._cached_android_proto_size_in_bytes = self.get_proto_size()
+            self.android_proto_size_is_stale = False
+
+        return self._cached_android_proto_size_in_bytes
+
+    def __setattr__(self, attrname, new_value):
+        if (attrname != '_cached_android_proto_size_in_bytes' and
+            attrname != 'proto_size_in_bytes' and
+            attrname != 'android_proto_size_is_stale'):
```

Comment:
```suggestion
        if attrname not in (
            '_cached_android_proto_size_in_bytes',
            'proto_size_in_bytes',
            'android_proto_size_is_stale'
        ):
```


and maybe put the three variable names into some constant? Also, explain why the condition is here in a code comment.

Reply:
Fixed